src/libs/runtime-utils.js

```diff
@@ -241,43 +241,55 @@ export const getCurrentRuntime = runtimes => {
   return !runtimes ? undefined : (_.flow(trimRuntimesOldestFirst, _.last)(runtimes) || null)
 }
 
-export const trimAppsOldestFirst = _.flow(
-  _.remove({ status: 'DELETING' }),
-  _.remove({ status: 'PREDELETING' }),
-  _.sortBy('auditInfo.createdDate'))
-
-export const getCurrentAppForType = appType => _.flow(trimAppsOldestFirst, _.filter(app => app.appType === appType), _.last)
+const getCurrentAppExcludingStatuses = (appType, statuses) => _.flow(
+  _.filter({ appType }),
+  _.remove(({ status }) => _.includes(status, statuses)),
+  _.sortBy('auditInfo.createdDate'),
+  _.last
+)
 
-export const currentAppIncludingDeleting = appType => _.flow(_.filter(app => app.appType === appType), _.sortBy('auditInfo.createdDate'), _.last)
+export const getCurrentApp = appType => getCurrentAppExcludingStatuses(appType, ['DELETING', 'PREDELETING'])
+export const getCurrentAppIncludingDeleting = appType => getCurrentAppExcludingStatuses(appType, [])
 
-export const currentAttachedDataDisk = (app, appDataDisks) => {
+export const getCurrentAttachedDataDisk = (app, appDataDisks) => {
   return _.find({ name: app?.diskName }, appDataDisks)
 }
 
+// If the disk was attached to an app, return the appType. Otherwise return undefined.
+export const getDiskAppType = disk => {
+  const saturnApp = disk.labels.saturnApplication
+  // Do a case-insensitive match as disks have been created with both "galaxy" and "GALAXY".
+  const appType = _.find(type => type.toLowerCase() === saturnApp?.toLowerCase(), getAllAppTypes)
+  return appType
```

Comment:
Especially since the function name nicely describes what it'd return:
```suggestion
  return _.find(type => type.toLowerCase() === saturnApp?.toLowerCase(), getAllAppTypes)
```